src/test/java/org/mule/test/oauth2/internal/clientcredentials/ClientCredentialsTokenTestCase.java

```diff
@@ -219,13 +226,72 @@ public void clientCredentialsInQueryParams() throws Exception {
 
     assertThat(requestCaptor.getValue().getHeaderNames(), not(hasItem(equalToIgnoringCase(AUTHORIZATION))));
 
-    assertThat(requestCaptor.getValue().getQueryParams().get("client_id"), is("Aladdin"));
-    assertThat(requestCaptor.getValue().getQueryParams().get("client_secret"), is("openSesame"));
+    assertThat(requestCaptor.getValue().getQueryParams().get(CLIENT_ID), is(ALADDIN));
+    assertThat(requestCaptor.getValue().getQueryParams().get(CLIENT_SECRET), is(OPEN_SESAME));
 
     String requestBody = IOUtils.toString(requestCaptor.getValue().getEntity().getContent(), UTF_8);
     assertThat(requestBody, containsString("grant_type=client_credentials"));
     assertThat(requestBody, not(containsString("client_secret=openSesame")));
     assertThat(requestBody, not(containsString("client_id=Aladdin")));
   }
 
+  @Test
+  public void clientCredentialsWithCustomQueryParams() throws Exception {
+    final OAuthClientCredentialsDancerBuilder builder = baseClientCredentialsDancerBuilder();
+    builder.tokenUrl("http://host/token");
+    builder.clientCredentials(ALADDIN, OPEN_SESAME);
+    builder.withClientCredentialsIn(QUERY_PARAMS);
+
+    MultiMap<String, String> queryParams = new MultiMap<>();
+
+    final String daenerys = "Daenerys";
+    final String[] daenerysValues = new String[] {"First of her name", "Mother of Dragons", "Mad Queen"};
+    final String jonSnow = "Jon Snow";
+    final String[] snowValues = new String[] {"Commander of the Night Watch", "Looser"};
```

Comment:
why looser? :(

Reply:
because he lost every single battle. The last three of them were actually won by his little sisters and his girlfriend